lib/reply.js

```diff
@@ -133,15 +133,21 @@ function onSendEnd (reply, payload) {
     reply.res.setHeader('Content-Type', 'text/plain')
   } else if (reply._serializer) {
     payload = reply._serializer(payload)
+    if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
+      throw new TypeError(`Serializer for Content-Type '${contentType}' returned invalid payload of type '${typeof payload}'. Expected a string or Buffer.`)
+    }
   } else if (!contentType || contentType === 'application/json') {
     reply.res.setHeader('Content-Type', 'application/json')
     payload = serialize(reply.context, payload, reply.res.statusCode)
     flatstr(payload)
+  } else if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
```

Comment:
`typeof payload` is run twi times, one on the first if and one here: https://github.com/fastify/fastify/pull/596/files#diff-d2fdf63c1d7f1d9f68270b6dfa8c930cL132.

Can you please reuse this?

Reply:
Yes

Reply:
Done